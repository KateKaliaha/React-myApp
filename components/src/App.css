@import './component/Header/Header.css';
@import './component/Search/Search.css';
@import './pages/MainPage/MainPage.css';
@import './component/Card/Card.css';
@import './component/CardList/CardList.css';
@import './component/Message/Message.css';
@import './pages/AboutPage/About.css';
@import './pages/ErrorPage/Error.css';
@import './pages/FormPage/FormPage.css';

* {
  box-sizing: border-box;
}

html {
  height: 100%;
}

* {
  box-sizing: border-box;
}

html {
  height: 100%;
}

.App {
  text-align: center;
}

body {
  background-color: #e9e9e9;
}
