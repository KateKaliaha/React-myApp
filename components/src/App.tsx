import { RouterInApp } from 'component/Router/Router';
import React from 'react';
import './App.css';

function App() {
  return <RouterInApp />;
}

export default App;
