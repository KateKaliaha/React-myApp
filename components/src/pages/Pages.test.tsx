import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from 'App';
import userEvent from '@testing-library/user-event';

describe('App', () => {
  afterEach(cleanup);

  it('snapshot home page', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('render main page', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    const linkMain = screen.getByText(/Главная/i);
    expect(linkMain).toBeInTheDocument();
    expect(screen.queryByText(/Страница не найдена/i)).not.toBeInTheDocument();
    expect(screen.queryByRole('searchbox')).not.toBeInTheDocument();
    userEvent.click(linkMain);
    expect(screen.queryByRole('searchbox')).toBeInTheDocument();
    expect(screen.queryByText(/Перейти на главную страницу/i)).not.toBeInTheDocument();
  });

  it('render form page', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    const linkForm = screen.getByText(/Отзыв/i);
    expect(linkForm).toBeInTheDocument();
    userEvent.click(linkForm);
    const formPage = screen.getByTestId('form-page');
    expect(formPage).toBeInTheDocument();
    expect(screen.queryByText(/Страница не найдена/i)).not.toBeInTheDocument();
  });
});
