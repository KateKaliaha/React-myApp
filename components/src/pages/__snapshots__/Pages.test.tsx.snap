// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot home page 1`] = `
<div>
  <header
    class="header"
    style="display: flex;"
  >
    <nav
      class="nav-container"
    >
      <a
        class=""
        href="/movie"
      >
        Главная
      </a>
      <a
        class=""
        href="/about"
      >
        О нас
      </a>
      <a
        class=""
        href="/form"
      >
        Отзыв
      </a>
    </nav>
  </header>
  <main
    class="container"
  >
    <div
      class="home-page"
    >
      <p>
        Добро пожаловать на наш сайт!
      </p>
      <a
        href="/movie"
      >
        Перейти на главную страницу
      </a>
    </div>
  </main>
</div>
`;
